types/src/proof/position/mod.rs

```diff
@@ -233,7 +233,7 @@ fn turn_off_right_most_n_bits(v: u64, n: u32) -> u64 {
 /// ```
 /// http://www.catonmat.net/blog/low-level-bit-hacks-you-absolutely-must-know/
 fn isolate_rightmost_zero_bit(v: u64) -> u64 {
-    !v & (v + 1)
+    !v & v.overflowing_add(1).0
```

Comment:
// v & (v+1) but hack to deal with overflow

Reply:
it's not a hack : o